app/services/AuditService.scala

```diff
@@ -124,8 +124,8 @@ class AuditService @Inject() (auditConnector: AuditConnector)(implicit ec: Execu
     val enrollmentReasons =
       if (eligibilityCheckResult.isEligible) None else Some(EnrollmentReasons.DidNotPassEligibilityCheck())
     val eligibilityReasons = {
-      val reasons = eligibilityCheckResult.eligibilityRules.getClass.getDeclaredFields.map(_.getName)
-      val values = eligibilityCheckResult.eligibilityRules.productIterator.to
+      val reasons: Array[String] = eligibilityCheckResult.eligibilityRules.getClass.getDeclaredFields.map(_.getName)
+      val values = eligibilityCheckResult.eligibilityRules.productIterator.iterator
```

Comment:
Compiler made me do this, I don't think there's any breaking change to this functionality, as we have tests for it